components/collections/src/codepointinvliststringlist/mod.rs

```diff
@@ -0,0 +1,307 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module provides functionality for querying of sets of Unicode code points and strings.
+//!
+//! It depends on [`CodePointInversionList`] to efficiently represent Unicode code points, while
+//! it also maintains a list of strings in the set.
+//!
+//! It is an implementation of the the existing [ICU4C UnicodeSet API](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1UnicodeSet.html).
+
+use crate::codepointinvlist::{CodePointInversionList, CodePointInversionListError};
+use alloc::string::{String, ToString};
+use displaydoc::Display;
+use yoke::Yokeable;
+use zerofrom::ZeroFrom;
+use zerovec::VarZeroVec;
+
+/// A data structure providing a concrete implementation of a `UnicodeSet`
+/// (which represents a set of code points and strings) using an inversion list for the code points and a simple
+/// list-like structure to store and iterate over the strings.
+#[derive(Debug, Eq, PartialEq, Clone, Yokeable, ZeroFrom)]
+// Valid to auto-derive Deserialize because the invariants are weakly held
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct CodePointInversionListStringList<'data> {
```

Comment:
So this is explicitly a struct of two separate parts.
I suggest adding "And" into the name: CodePointInversionListAndStringList

Reply:
Done.